entities/player/Player.gd

```diff
@@ -38,17 +43,21 @@ func _physics_process(_delta):
 	_process_collisions()
 	move_and_slide()
 
+func damage_player(amount):
```

Comment:
Okay this function looks good. I have it named handle_hit and it simply reduces the value by 1. And I created a custom setter function for health so we can move the code for is_dead there.

Is it possible for you to combine this invincibility code with something like on_hit the player can't take more damage for 0.5 like you did invincibility?

Reply:
I'm not fully sure what you mean but I can make the invincibility code work with whatever function you make, once you merge it I'll figure it out